gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterMetricsStatsExporter.java

```diff
@@ -0,0 +1,145 @@
+        this.refreshInterval = monitorConfiguration.getClusterMetricsRegistryRefreshPeriod();
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        log.debug("Running periodic metric refresh with interval of %s", refreshInterval);
+        scheduledExecutor.scheduleAtFixedRate(() -> {
+            try {
+                updateClustersMetricRegistry();
+            }
+            catch (Exception e) {
+                log.error(e, "Error refreshing cluster metrics");
+            }
+        }, 0, refreshInterval.toMillis(), MILLISECONDS);
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        scheduledExecutor.shutdownNow();
+    }
+
+    private synchronized void updateClustersMetricRegistry()
+    {
+        // Get current clusters from DB
+        Set<String> currentClusters = gatewayBackendManager.getAllBackends().stream()
+                .map(ProxyBackendConfiguration::getName)
+                .collect(Collectors.toSet());
+
+        // Create a copy of keys to avoid concurrent modification
+        Set<String> registeredClusters = new HashSet<>(clustersStats.keySet());
+
+        // Unregister metrics for removed clusters
+        for (String registeredCluster : registeredClusters) {
+            if (!currentClusters.contains(registeredCluster)) {
+                try {
+                    exporter.unexportWithGeneratedName(ClusterMetricsStats.class, registeredCluster);
+                    log.debug("Unregistered metrics for removed cluster: %s", registeredCluster);
+                    clustersStats.remove(registeredCluster);
+                }
+                catch (Exception e) {
+                    log.error(e, "Failed to unregister metrics for cluster: %s", registeredCluster);
+                }
+            }
+        }
+
+        // Register metrics for added clusters
+        for (String cluster : currentClusters) {
+            if (!clustersStats.containsKey(cluster)) {
+                registerClusterMetrics(cluster);
+            }
+        }
+    }
+
+    private synchronized void registerClusterMetrics(String clusterName)
+    {
+        ClusterMetricsStats stats = new ClusterMetricsStats(clusterName, gatewayBackendManager);
+
+        if (clustersStats.putIfAbsent(clusterName, stats) == null) {  // null means the stats didn't exist previously and was inserted
+            try {
+                exporter.exportWithGeneratedName(stats, ClusterMetricsStats.class, clusterName);
+                log.debug("Registered metrics for cluster: %s", clusterName);
+            }
+            catch (Exception e) {
+                clustersStats.remove(clusterName);
+                log.error(e, "Failed to register metrics for cluster: %s", clusterName);
+            }
+        }
+        else {
+            log.warn("Attempted to register metrics for duplicate cluster name: %s. This may cause JMX registration issues.", clusterName);
+        }
+    }
+
+    GatewayBackendManager getGatewayBackendManager()
```

Comment:
Please consider adding `@VisibleForTesting` and avoid `get` prefix. 
https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-get-in-method-names-unless-an-object-must-be-a-java-bean